gtp2ogs.js

```diff
@@ -306,7 +306,7 @@ class Game {
                 if (this.waiting_on_gamedata_to_make_move && this.state.moves.length == this.move_number_were_waiting_for) {
                     this.makeMove(this.move_number_were_waiting_for);
                     this.waiting_on_gamedata_to_make_move = false;
-                    this.move_number_were_waiting_for = -1;
+                    this.move_number_were_waiting_for += 2;
                 }
```

Comment:
By keeping the move we're waiting for current at all times, persistent bots can receive move events properly even if the active_game event comes in after the move.